### DISCUSSION TITLE
Migrating from docker: running hadolint

### DISCUSSION THREAD

**Discussion by wwuck** [containers/podman#12552] [Q&A]
Docker version: 5:20.10.11~3-0~debian-bullseye
Podman version: 3.0.1+dfsg1-3+b2

I'm trying to migrate from docker and having into a problem when trying to run [hadolint](https://github.com/hadolint/hadolint/) docker image.

When I run with `docker`, it is successful:
```console
$ docker run -it docker.io/hadolint/hadolint:v2.8.0 hadolint
Please provide a Dockerfile
```
However when I run with `podman` I get an error:
```console
$ podman run -it docker.io/hadolint/hadolint:v2.8.0 hadolint 
hadolint: getXdgDirectory:getHomeDirectory:getUserEntryForID: does not exist (No such file or directory)
```
Is this some configuration I need to change for podman, or is it a change needed in hadolint image to run on both docker and podman, or something else?

**Comment by rhatdan** [comment]
Worked for me.
```
$ podman run -it docker.io/hadolint/hadolint:v2.8.0 hadolint 
Trying to pull docker.io/hadolint/hadolint:v2.8.0...
Getting image source signatures
Copying blob d0727728368a done  
Copying config 2a94a15d7d done  
Writing manifest to image destination
Storing signatures
Please provide a Dockerfile
```

Looks like you might have some strangeness going on. With getXdgDirectory?

**Reply by rhatdan** [reply]
$ podman -v
podman version 3.4.1

**Reply by wwuck** [reply]
I'm running podman 3.0.1+dfsg1-3+b2 on Devuan 4/chimaera (Debian 11/bullseye derivative, the packages come direct from Debian repository).  I just uninstalled docker-ce and ran `apt-get install podman`.
The only configuration change I made was to set this in `containers.conf`:
```ini
[engine]
cgroup_manager = "cgroupfs"
events_logger = "file"
```

Unfortunately I can't test the newer version from `kubic` repository due to it's dependency on `dbus-user-session` which is unavailable in Devuan.  This dependency does not exist in the Debian package archive version.

Are there changes between 3.0.1 and 3.4.1 that would affect this?

**Reply by wwuck** [reply]
I just tested also on a fresh VM for Debian Testing and I get the same error.  I just installed from the latest debian-testing netinst ISO image, unselected the GUI task selection (GNOME, etc.) in installer to get a console-only install and then ran `apt-get update && apt-get install podman` from the root console.

This is running podman 3.4.2.  Could it be an issue with the way Debian has packaged podman?

**Reply by wwuck** [reply]
Also, `getXdgDirectory:getHomeDirectory:getUserEntryForID` is from Haskell so that is the error coming from hadolint, not from podman.

**Reply by wwuck** [reply]
I can get it working if I run the `hadolint-debian` or `hadolint-alpine` images which use `FROM debian:stretch-slim` and `FROM alpine:3`.  The default `hadolint` image is using `FROM scratch`.

I'm guessing this is the source Dockerfile: https://github.com/hadolint/hadolint/blob/master/docker/Dockerfile

**Reply by wwuck** [reply]
I ran another test using latest Debian 11 plus the upstream kubic repository https://build.opensuse.org/package/show/devel:kubic:libcontainers:stable/podman to install podman 3.4.2 and that is working without problem.  This is leading me to think there is either a problem with the Debian packaging for podman or a configuration setting that is missing from the default Debian podman install but is present with the upstream Kubic apt package install?

**Comment by baude** [comment]
works with 4.0dev too

**Comment by ru5j4r0** [comment]
It's been half a year and maybe you already solved this issue, but I share my workaround without Kubic's one, at least for future viewers.
I'm using Ubuntu 21.10.

`podman run -e 'HOME=/' --rm -i docker.io/hadolint/hadolint:latest hadolint`

I guess Hadolint requires a HOME environment variable and the environment variable of scratch image is handled differently depends on host environment or how Podman was built?
I didn't test with Kubic's build nor check what is happening inside, so I'm not sure about my guess though.
